src/utils/currency/contractPrice.ts

```diff
@@ -0,0 +1,99 @@
+import { Network, SupportedCurrencies } from '~gql';
+
+import { currencyApiConfig, coinGeckoMappings } from './config.ts';
+import {
+  type FetchCurrentPriceArgs,
+  type CoinGeckoSupportedCurrencies,
+  type SupportedChains,
+  type TokenContractPriceSuccessResponse,
+  type TokenContractPriceResponse,
+} from './types.ts';
+import { buildAPIEndpoint, fetchData, mapToAPIFormat } from './utils.ts';
+
+// The functions defined in this file assume something about the shape of the api response.
+// If that changes, or if we change the api, these functions will need to be updated.
+
+const { chains, currencies } = coinGeckoMappings;
+const buildTokenContractCoinGeckoURL = (
+  contractAddress: string,
+  chainId: SupportedChains = Network.Gnosis,
+  conversionDenomination: CoinGeckoSupportedCurrencies = SupportedCurrencies.Usd,
+) => {
+  const chain = mapToAPIFormat(chains, chainId);
+  const denomination = mapToAPIFormat(currencies, conversionDenomination);
+  const { url, searchParams } =
+    currencyApiConfig.endpoints.tokenPriceByContract;
+
+  return buildAPIEndpoint(new URL(`${url}/${chain}`), {
+    [searchParams.from]: contractAddress,
+    [searchParams.to]: denomination,
+    [searchParams.api]: process.env.COINGECKO_API_KEY ?? '',
+  });
+};
+
+const extractContractPriceFromResponse = (
+  data: TokenContractPriceSuccessResponse,
+  contractAddress: string,
+  conversionDenomination: CoinGeckoSupportedCurrencies,
+) => {
+  try {
+    return data[contractAddress.toLowerCase()][
```

Comment:
Why `toLowerCase` and not checksumming? Is it coming as a lowercase value from coingecko ?

Reply:
I just copied this over, and double checked that yes, CoinGecko does actually return everything in lower case
should we rather do checksumming here?

Reply:
Nope. If they return lowercase, then you've handled it here correctly.